### PR TITLE
Crushin Canz 💯 

### PR DESCRIPTION
**Pull request by kugamo**
## About The Pull Request

You can now crush empty cans! 

With your head! _(attack self with the can while targeting head on harm intent)_

With your hands! _(while can in inactive hand click with empty active hand on harm intent)_

With your foot! _(click on the can on the ground while targeting foot on harm intent)_

## Why It's Good For The Game

Mindless fun. Impress your squadmates with your bravado!

https://user-images.githubusercontent.com/16618648/205354207-3e8d95ff-ba09-4eb6-80af-8b2913caeb41.mp4

## Changelog

:cl:
add: You can now crush cans on harm intent with your head, hands, or feet!
soundadd: New sound for crushin cans.
imageadd: New sprites for crushed cans.
/:cl:

### CODE REVIEW

**Review by Agameofscones — APPROVED** [review]
sprites look good, sound is fine

**Comment by Pyrodes** [comment]
soul

File: code/game/objects/items/reagent_containers/food/cans.dm [hunk 1]
```diff
@@ -1,28 +1,54 @@
...
 	gulp_size = 10
+	icon = 'icons/obj/items/drinkcans.dmi'
 
 /obj/item/reagent_container/food/drinks/cans/attack_self(mob/user)
 	..()
 
+	if(crushed)
+		return
+
 	if (canopened == FALSE)
```
Comment (Shad0vvs) [inline code/game/objects/items/reagent_containers/food/cans.dm:13]:
```suggestion
	if (!canopened)
```

File: code/game/objects/items/reagent_containers/food/cans.dm [hunk 2]
```diff
@@ -1,28 +1,54 @@
 /obj/item/reagent_container/food/drinks/cans
 	var/canopened = 0
```
Comment (Shad0vvs) [inline code/game/objects/items/reagent_containers/food/cans.dm:2]:
```suggestion
	var/canopened = FALSE
```

File: code/game/objects/items/reagent_containers/food/cans.dm [hunk 3]
```diff
@@ -1,28 +1,54 @@
...
 	if (canopened == FALSE)
 		playsound(src.loc,'sound/effects/canopen.ogg', 15, 1)
 		to_chat(user, SPAN_NOTICE("You open the drink with an audible pop!"))
 		canopened = TRUE
 
+/obj/item/reagent_container/food/drinks/cans/attack_hand(mob/user)
+	if(crushed)
+		return ..()
+
+	if (canopened == 1 && !reagents.total_volume)
```
Comment (Shad0vvs) [inline code/game/objects/items/reagent_containers/food/cans.dm:22]:
```suggestion
	if (canopened && !reagents.total_volume)
```

File: code/game/objects/items/reagent_containers/food/cans.dm [hunk 4]
```diff
@@ -1,28 +1,54 @@
...
+			else if(loc == user && src == user.get_inactive_hand())
+				crush_can(user)
+				return FALSE
+	return ..()
+
 /obj/item/reagent_container/food/drinks/cans/attack(mob/M, mob/user)
+	if(crushed)
+		return
+
 	if (canopened == 0)
```
Comment (Shad0vvs) [inline code/game/objects/items/reagent_containers/food/cans.dm:37]:
```suggestion
	if (!canopened)
```

File: code/game/objects/items/reagent_containers/food/cans.dm [hunk 5]
```diff
@@ -34,7 +60,6 @@
 			to_chat(user, SPAN_NOTICE("You need to open the drink!"))
```
Comment (Shad0vvs) [inline code/game/objects/items/reagent_containers/food/cans.dm:60]: The one above this line I cant fix 💀

File: code/game/objects/items/reagent_containers/food/cans.dm [hunk 6]
```diff
@@ -90,7 +115,41 @@
```
Comment (Shad0vvs) [inline code/game/objects/items/reagent_containers/food/cans.dm:115]: 3 above here just to make them good 💀

Comment (kugamo) [reply code/game/objects/items/reagent_containers/food/cans.dm:115]: 💀

**Review by Shad0vvs — APPROVED** [review]

**Pull request merged**